mobile/src/main/java/rs/readahead/washington/mobile/views/fragment/reports/viewpagerfragments/SubmittedReportsFragment.kt

```diff
@@ -0,0 +1,108 @@
+    FragmentReportsListBinding::inflate
+) {
+    private val viewModel by viewModels<ReportsEntryViewModel>()
+    private val entityAdapter: EntityAdapter by lazy { EntityAdapter() }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initView()
+        initData()
+    }
+
+    private fun initView() {
+        binding?.textViewEmpty?.setText(getString(R.string.Submitted_Reports_Empty_Message))
+        binding?.listReportsRecyclerView?.apply {
+            adapter = entityAdapter
+            layoutManager = LinearLayoutManager(baseActivity)
+        }
+    }
+
+    private fun initData() {
+        with(viewModel) {
+            submittedReportListFormInstance.observe(viewLifecycleOwner) { outboxes ->
+                if (outboxes.isEmpty()) {
+                    binding?.listReportsRecyclerView?.hide()
+                    binding?.textViewEmpty?.show()
+                } else {
+                    entityAdapter.setEntities(outboxes)
+                    binding?.listReportsRecyclerView?.show()
+                    binding?.textViewEmpty?.hide()
+                }
+            }
+
+            onMoreClickedFormInstance.observe(viewLifecycleOwner) { instance ->
+                showSubmittedMenu(instance)
+            }
+
+            reportInstance.observe(viewLifecycleOwner) { instance ->
+                openEntityInstance(instance)
+            }
+
+            onOpenClickedFormInstance.observe(viewLifecycleOwner) { instance ->
+                loadEntityInstance(instance)
+            }
+
+            instanceDeleted.observe(viewLifecycleOwner) {
+                viewModel.listOutbox()
+            }
+        }
+    }
+
+    private fun showSubmittedMenu(instance: ReportFormInstance) {
+        BottomSheetUtils.showEditDeleteMenuSheet(
+            requireActivity().supportFragmentManager,
+            instance.title,
+            getString(R.string.View_Report),
+            getString(R.string.Delete_Report),
+            object : BottomSheetUtils.ActionSeleceted {
+                override fun accept(action: BottomSheetUtils.Action) {
+                    if (action === BottomSheetUtils.Action.EDIT) {
+                        loadEntityInstance(instance)
+                    }
+                    if (action === BottomSheetUtils.Action.DELETE) {
+                        viewModel.deleteReport(instance)
+                    }
+                }
+            },
+            getString(R.string.action_delete) + " \"" + instance.title + "\"?",
+            requireContext().resources.getString(R.string.Delete_Submitted_Report_Confirmation),
+            requireContext().getString(R.string.action_yes),
+            requireContext().getString(R.string.action_no)
+        )
+    }
+
+    private fun loadEntityInstance(reportFormInstance: ReportFormInstance) {
+        viewModel.getDraftBundle(reportFormInstance)
```

Comment:
We should rename this fun getDraftBundle for something like getReportBundle. In the end, it is used in many places, not just Drafts so the name is misleading.